app/controllers/discovery_rules_controller.rb

```diff
@@ -8,7 +8,7 @@ class DiscoveryRulesController < ApplicationController
 
   def index
     base = resource_base.search_for(params[:search], :order => (params[:order]))
-    @discovery_rules = base.paginate(:page => params[:page]).includes(:hostgroup)
+    @discovery_rules = base.paginate(:page => params[:page], :per_page => params[:per_page]).includes(:hostgroup)
```

Comment:
why not use `resource_base_search_and_page` helper which handles all of these parameters?

Reply:
We have another patch that touches this part https://github.com/theforeman/foreman_discovery/pull/406

Reply:
I remember we had issues because discovery rules page contains search in different context. It did not work out.

Reply:
@lzap yes, the search field disappears when using `resource_base_search_and_page`. @tbrisker can we think of something to avoid the same?

Reply:
That's very strange, since in core we have:
```ruby
  def resource_base_with_search
    resource_base.search_for(params[:search], :order => params[:order])
  end

  def resource_base_search_and_page(tables = [])
    base = tables.empty? ? resource_base_with_search : resource_base_with_search.eager_load(*tables)
    base.paginate(:page => params[:page], :per_page => params[:per_page])
  end
```
which is exactly the same...